### DISCUSSION TITLE
How to automatically split a large CSV file into smaller files?

### DISCUSSION THREAD

**Discussion by herbst010** [saulpw/visidata#2540] [Q&A] [answered]
Hello everyone, I have a single .csv file with 8 million rows, and I'd like to split it into multiple .csv files of 1 million rows each, automatically saving them locally on .csv format. Is it possible to do this with a Python script on Visidata? If so, could you help me create it?

**Comment by saulpw** [comment, marked as answer]
There are better tools than VisiData to do this for csv, assuming none of the fields have newlines in them.  Use the POSIX tools `head` and `split`.

**Reply by frosencrantz** [reply]
Miller can do this: https://miller.readthedocs.io/en/latest/reference-verbs/#split
